### PR TITLE
Omni search

### PR DESCRIPTION
**Pull request by richtier**

### CODE REVIEW

File: caseworker/search/views.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        context["results"] = results
+        context["data"] = {"total_pages": results["count"] // form.page_size}
+        return context
+
+
+class AutocompleteView(views.APIView):
+    authentication_classes = []
+    permission_classes = [
+        permissions.AllowAny,
```
Comment (r4vi) [inline caseworker/search/views.py:39]: shouldn't we check they're a logged in user?

**Review by r4vi — APPROVED** [review]
looks good, just some files missing newlines

File: caseworker/search/views.py (hunk 1 above)
Comment (r4vi) [reply caseworker/search/views.py:39]: maybe auth for all of this is something to resolve later - potentially sending ES queries through the api layer too

File: caseworker/assets/css/components/_search.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+body {
+  #autoComplete_list {
```
Comment (eadpearce) [inline caseworker/assets/css/components/_search.scss:2]: can we stick to snake case in css class names pls and thank. you'll need to change them in the JS selectors as well

File: caseworker/templates/search/search.html [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                                <span class="govuk-visually-hidden">View</span> {{ item.id }}
+                            </a>
+                            <p class="govuk-body govuk-!-margin-top-2 govuk-!-margin-bottom-0">
+                                {{ item.organisation }}
+                            </p>
+                            <p class="govuk-tag govuk-tag--grey govuk-!-margin-0 govuk-!-margin-top-2">
+                                {{ item.status }}
+                            </p>
+                            <p>
+                                <details>
```
Comment (eadpearce) [inline caseworker/templates/search/search.html:62]: details tag inside a p is probably redundant. if you're using the p for extra margin just use one of the govuk margin classes

**Review by eadpearce — DISMISSED** [review]
small stuff

File: caseworker/assets/css/components/_search.scss (hunk 2 above)
Comment (eadpearce) [reply caseworker/assets/css/components/_search.scss:2]: (if this is a fault of the library we're using i guess i'll just have to direct my frustration at them instead)

Comment (saruniitr) [reply caseworker/assets/css/components/_search.scss:2]: These are only used in this file, changing their names to `autoCompleteList` affecting the styling, cannot find JS selectors

File: caseworker/templates/search/search.html (hunk 3 above)
Comment (saruniitr) [reply caseworker/templates/search/search.html:66]: fixed

File: caseworker/search/views.py (hunk 1 above)
Comment (richtier) [reply caseworker/search/views.py:39]: :+1: this was to get up and running quickly for dev. will add auth check

File: caseworker/assets/css/components/_search.scss (hunk 2 above)
Comment (richtier) [reply caseworker/assets/css/components/_search.scss:2]: that is indeed defined by the js lib and it's hard-coded too https://github.com/TarekRaafat/autoComplete.js/search?q=autoComplete_list&unscoped_q=autoComplete_list

**Pull request merged**